packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2956,11 +2996,9 @@ export class SnapController extends BaseController<
    * @returns The metadata for the given snap ID, or `null` if the snap is not
    * verified.
    */
-  async getRegistryMetadata(
-    snapId: SnapId,
-  ): Promise<SnapsRegistryMetadata | null> {
+  getRegistryMetadata(snapId: SnapId): SnapsRegistryMetadata | null {
     this.#assertCanUsePlatform();
-    return await this.messagingSystem.call('SnapsRegistry:getMetadata', snapId);
+    return this.messagingSystem.call('SnapsRegistry:getMetadata', snapId);
   }
```

Comment:
Honestly this function should probably be deleted, I have no idea why you would want to call `SnapController:getRegistryMetadata` when you can just use the registry directly

Reply:
Oh I don't mean why you specifically would do it for this PR. I mean why would anyone call this function that we expose 😂